src/combinatorics/simple_graph/basic.lean

```diff
@@ -332,4 +332,32 @@ finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty
 
 end finite
 
+section common_neighbors
+
+/--
+The set of common neighbors between two vertices `v` and `w` in a graph `G` is the
+intersection of the neighbor sets of `v` and `w`.
+-/
+def common_neighbors (v w : V) : set V := (G.neighbor_set v) ∩ (G.neighbor_set w)
+
+lemma common_neighbors_eq_inter (v w : V) :
+  G.common_neighbors v w = (G.neighbor_set v) ∩ (G.neighbor_set w) := rfl
+
+lemma not_mem_left_common_neighbors (v w : V) : v ∉ (G.common_neighbors v w) :=
+by simp [irrefl, mem_neighbor_set, common_neighbors_eq_inter]
+
+lemma not_mem_right_common_neighbors (v w : V) : w ∉ (G.common_neighbors v w) :=
+begin
+  rw common_neighbors,
+  simp,
+end
+
+lemma common_neighbors_subset_neighbor_set (v w : V) : G.common_neighbors v w ⊆ G.neighbor_set v :=
+begin
+  rw common_neighbors_eq_inter,
+  exact set.inter_subset_left _ _,
+end
+
+end common_neighbors
+
```

Comment:
There are some parentheses you can remove (around each `G.common_neighbors v w`).  You can rename `common_neighbors_eq_inter` to `common_neighbors_eq` for short. It might be useful having `common_neighbors_symm`.

Many of these can be proved by `by simp [common_neighbors]`:

```lean
def common_neighbors (v w : V) : set V := G.neighbor_set v ∩ G.neighbor_set w

lemma common_neighbors_eq (v w : V) :
  G.common_neighbors v w = G.neighbor_set v ∩ G.neighbor_set w := rfl

lemma common_neighbors_symm (v w : V) : G.common_neighbors v w = G.common_neighbors w v :=
by { rw [common_neighbors, set.inter_comm], refl }

lemma not_mem_left_common_neighbors (v w : V) : v ∉ G.common_neighbors v w :=
by simp [common_neighbors]

lemma not_mem_right_common_neighbors (v w : V) : w ∉ G.common_neighbors v w :=
by simp [common_neighbors]

lemma common_neighbors_subset_neighbor_set (v w : V) : G.common_neighbors v w ⊆ G.neighbor_set v :=
by simp [common_neighbors]
```
I wonder what the mathlib name for `not_mem_left_common_neighbors` would be.  Maybe `not_mem_common_neighbors_left` might be better? I'm not sure.